python/lsst/ts/watcher/base_hexapod_overcurrent_rule.py

```diff
@@ -0,0 +1,190 @@
+# This file is part of ts_watcher.
+#
+# Developed for Vera C. Rubin Observatory Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["BaseHexapodOvercurrentRule"]
+
+import logging
+import types
+import typing
+
+import yaml
+from lsst.ts import salobj
+from lsst.ts.xml.enums.MTHexapod import ControllerState, EnabledSubstate, SalIndex
+from lsst.ts.xml.enums.Watcher import AlarmSeverity
+
+from .base_rule import AlarmSeverityReasonType, BaseRule, NoneNoReason
+from .remote_info import RemoteInfo
+
+
+class BaseHexapodOvercurrentRule(BaseRule):
+    """Base rule to monitor the main telescope hexapod overcurrent event.
+
+    Parameters
+    ----------
+    salIndex : enum `SalIndex`
+        Hexapod SAL index.
+    config : `types.SimpleNamespace`
+        Rule configuration, as validated by the schema.
+    log : `logging.Logger` or None, optional
+        Parent logger. (the default is None)
+    """
+
+    def __init__(
+        self,
+        salIndex: SalIndex,
+        config: types.SimpleNamespace,
+        log: logging.Logger | None = None,
+    ):
+        is_camera_hexapod = salIndex == SalIndex.CAMERA_HEXAPOD
```

Comment:
Please, consider making the index or, even better, the "name" of the hexapod (as in the `SalIndex` enumeration) a configuration parameter. Then you won't need to subclass it.

Reply:
Not sure I understand the use of SAL index here. Do you mean to use the `SalIndex` in the configuration parameter to say we want to have the M2 hexapod alarm or camera hexapod alarm? But I thought we should have two alarms for two hexapods together because:

1. Both of them would have the overcurrent.
2. To be able to mute the individual alarm.

Please help to clarify more. Maybe I do not understand your meaning.

Thanks!

Reply:
Updated in the commit: [70904dd](https://github.com/lsst-ts/ts_watcher/pull/94/commits/70904dd567affd77fdcbd259644d369adf303a71). Thanks!